src/bos/common/clients/bos/components.py

```diff
@@ -23,19 +23,31 @@
 #
 import logging
 
-from .base import BaseBosEndpoint
+from .base import BaseBosNonTenantAwareEndpoint
+from .options import options
 
 LOGGER = logging.getLogger(__name__)
 
 
-class ComponentEndpoint(BaseBosEndpoint):
+class ComponentEndpoint(BaseBosNonTenantAwareEndpoint):
     ENDPOINT = __name__.lower().rsplit('.', maxsplit=1)[-1]
 
     def get_component(self, component_id):
         return self.get_item(component_id)
 
     def get_components(self, **kwargs):
-        return self.get_items(**kwargs)
+        page_size = options.max_component_batch_size
```

Comment:
page size is define as constant here. Should it not come as part of REST request?

Reply:
I am not sure what you mean by part of the REST request -- this function is what ends up generating the REST requests.

But if you mean should it be a parameter to this function, then it's complicated. In theory, sure. The option value being used only applies to cases where a BOS operator is the one making the call into the function. So in theory, if something other than a BOS operator is using this client, then they may want to use a different value.

In practice, there are only two things that use these API clients -- the BOS server and the BOS operators. But for the BOS client specifically, the BOS server doesn't ever use it, since it has no need to -- it can just talk to itself directly. So in practice this client will only ever be called by an operator, in which case, this is the value we'd want to use for the page size.

That said, it wouldn't hurt to move it to be a function parameter, and just use that as the default value. It will amount to the same thing.

Reply:
Do you want to move it to function parameter in same PR or new PR?

Reply:
I already moved it in this PR